uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/staff-leave-report.component.js

```diff
@@ -200,32 +232,43 @@ define([
      * @return {Promise}
      */
     function loadBalanceChanges () {
+      var basicParams = { contact_id: vm.contactId, period_id: vm.selectedPeriod.id };
+
       return $q.all([
-        LeaveRequest.balanceChangeByAbsenceType(vm.contactId, vm.selectedPeriod.id, null, true),
-        LeaveRequest.balanceChangeByAbsenceType(vm.contactId, vm.selectedPeriod.id, [
-          valueOfRequestStatus(sharedSettings.statusNames.approved)
-        ]),
-        LeaveRequest.balanceChangeByAbsenceType(vm.contactId, vm.selectedPeriod.id, [
-          valueOfRequestStatus(sharedSettings.statusNames.awaitingApproval),
-          valueOfRequestStatus(sharedSettings.statusNames.moreInformationRequired)
-        ])
+        LeaveRequest.balanceChangeByAbsenceType(_.assign({}, basicParams, {
+          public_holiday: true
+        })),
+        LeaveRequest.balanceChangeByAbsenceType(_.assign({}, basicParams, {
+          expired: true
+        })),
+        LeaveRequest.balanceChangeByAbsenceType(_.assign({}, basicParams, {
+          statuses: {
+            in: [ valueOfRequestStatus(sharedSettings.statusNames.approved) ]
+          }
+        })),
+        LeaveRequest.balanceChangeByAbsenceType(_.assign({}, basicParams, {
+          statuses: {
+            in: [
+              valueOfRequestStatus(sharedSettings.statusNames.awaitingApproval),
+              valueOfRequestStatus(sharedSettings.statusNames.moreInformationRequired)
+            ]
+          }
+        }))
```

Comment:
```js
var balanceChangeRequestFilters = {
  holidays: { public_holiday: true },
  expired: { expired: true },
  approved: { statuses: { in: [ valueOfRequestStatus(sharedSettings.statusNames.approved) ] } },
  pending: {
    statuses: {
      in: [
        valueOfRequestStatus(sharedSettings.statusNames.awaitingApproval),
        valueOfRequestStatus(sharedSettings.statusNames.moreInformationRequired)
      ]
    }
};

var balanceChangeRequests = _.transform(balanceChangeRequestFilters, function (promises, filters, requestName) {
  promises[requestName] = LeaveRequest.balanceChangeByAbsenceType(_.assign({}, basicParams, filters);
}, {});

$q.all(balanceChangeRequests)
  .then(function (results) {
    vm.absenceTypes.forEach(function (absenceType) {
      absenceType.balanceChanges = {
        holiday: results.holiday[absenceType.id],
        // etc.
      };
      // if we somehow can make it so its like this it would be even better:
      absenceType.balanceChanges = results[absenceType.id];
    });
  });

```